Reserved typenames

hugomg opened the pull request.

mascarenhas committed:
fix 'and' and 'or' between non-boolean values when used in a value context instead of control context (fixes issue #121)

andremm committed:
Merge pull request #122 from titan-lang/issue121

fix issue #121 ('and' and 'or' between non-boolean values)

Commit:
Merge branch 'master' into lablua-hugo

Commit:
Clarify comment

Commit:
Basic type names are now reserved words

This lets us avoid a lot os silly nonsense in the typechecker.

The STRING lexer token now refers to the `string` keyword. The old
STRING token was renamed to STRINGLIT

gligneul commented on titan-compiler/checker.lua:
```diff
@@ -46,25 +46,51 @@ end
+        return types.Array(typefromnode(node.subtype, st, errors))
+    end,
+
+
```
s/\n//

gligneul commented on titan-compiler/checker.lua:
```diff
@@ -46,25 +46,51 @@ end
         local name = node.name
         local sym = st:find_symbol(name)
-        if sym then
+	    if sym then
```
Indentation is wrong here.

gligneul reviewed (changes requested):
Looks good, only minor details. Also, couldn't we remove this merge commit and make a rebase later.

Commit:
Indentation fixes

Commit:
type --> typ

Avoid creating local variables named "type", because they shadow an
existing function from the Lua standard library.
